addons/mail/static/src/components/message/message.xml

```diff
@@ -7,6 +7,7 @@
                 'o-clicked': state.isClicked,
                 'o-discussion': message and (message.is_discussion or message.is_notification),
                 'o-has-message-selection': threadView and threadView.threadViewer and threadView.threadViewer.selectedMessage,
+                'o-highlighted': message and message.isHighlighted and message.originThread and message.originThread.model === 'mail.channel',
```

Comment:
`message and message.isHighlighted` should be the only condition necessary.

`isHighlighted` should be computed to be correct depending on mentions and channel type.